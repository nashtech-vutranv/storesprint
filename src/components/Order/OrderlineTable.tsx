import {DataTable} from 'primereact/datatable'
import {useContext, useEffect, useState} from 'react'
import {useTranslation} from 'react-i18next'
import {Link} from 'react-router-dom'
import {Column} from 'primereact/column'
import {Button as ButtonPrime} from 'primereact/button'
import {useHandleError} from '../../hooks'
import OrderService from '../../services/OrderService'
import {GlobalContext} from '../../store/GlobalContext'
import FieldTextDataTable from '../FieldTextDataTable/FieldTextDataTable'
import {IOrderLine} from '../../interface/order'
import {ROUTE_PRODUCT, ROUTE_PARAMS} from '../../constants'

export interface IOrderlineTableProps {
  orderId?: string,
  getOrderDetail?: () => void
}

export default function OrderlineTable({orderId, getOrderDetail}: IOrderlineTableProps) {
  const {t} = useTranslation()
  const [orderLines, setOrderLines] = useState<IOrderLine[]>([])

  const {
    state: {axiosClient},
  } = useContext(GlobalContext)

  const {handleErrorResponse} = useHandleError()

  useEffect(() => {
    if (orderId) {
      getOrderLines()
    }
  }, [orderId])

  const getOrderLines = () => {
    new OrderService(axiosClient)
      .getOrderlines(`${orderId}`)
      .then((response: any) => {
        setOrderLines(response.data)
      })
      .catch((err: any) => {
        handleErrorResponse(err)
      })
  }
  
  const formatPrice = (price: string) => {
    return price.replace(/\.?0+$/, '')
  }

  const refreshData = () => {
    getOrderLines()
    if (getOrderDetail) {
      getOrderDetail()
    }
  }

  const paginatorLeft = (
    <ButtonPrime
      aria-label='refresh-button'
      type='button'
      icon='pi pi-refresh'
      className='p-button-text refresh-button'
      onClick={refreshData}
      tooltip={t('common_button_refresh_table')}
    />
  )

  return (
    <>
      <DataTable
        dataKey='id'
        value={orderLines}
        rows={orderLines.length}
        totalRecords={orderLines.length}
        responsiveLayout='scroll'
        selectionMode='single'
        className='data-table-mh mt-4'
        emptyMessage={t('order_detail_table_no_data')}
        paginator
        paginatorTemplate=''
        paginatorClassName='table-paginator'
        paginatorLeft={paginatorLeft}
        lazy
      >
        <Column
          header={t('order_detail_table_product_erp_id')}
          body={(orderLine: IOrderLine) => (
            <FieldTextDataTable
              value={orderLine.product?.erpId}
              placement='bottom'
            />
          )}
        ></Column>
        <Column
          header={t('order_detail_table_name')}
          body={(orderLine: IOrderLine) => (
            <Link
              to={ROUTE_PRODUCT.DETAIL.replace(
                ROUTE_PARAMS.PRODUCT_ID,
                orderLine.product?.id || ''
              )}
            >
              {orderLine.product?.name}
            </Link>
          )}
        ></Column>
        <Column
          header={t('order_detail_table_quantity')}
          body={(orderLine: IOrderLine) => (
            <FieldTextDataTable value={orderLine.quantity} placement='bottom' />
          )}
        ></Column>
        <Column
          header={t('order_detail_table_price')}
          body={(orderLine: IOrderLine) => (
            <FieldTextDataTable value={formatPrice(orderLine.price)} placement='bottom' />
          )}
        ></Column>
        <Column
          header={t('order_detail_table_tax')}
          body={(orderLine: IOrderLine) => (
            <FieldTextDataTable
              value={orderLine.taxAppliedPerUnit}
              placement='bottom'
            />
          )}
        ></Column>
        <Column
          header={t('order_detail_table_orderline_status')}
          body={(orderLine: IOrderLine) => (
            <FieldTextDataTable
              value={orderLine.orderlineStatus}
              placement='bottom'
            />
          )}
        ></Column>
      </DataTable>
    </>
  )
}
