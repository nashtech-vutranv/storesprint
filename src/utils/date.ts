import moment from 'moment'
import {FORMAT_DATE, FORMATE_DATE_NO_HOURS} from '../constants'

export const formatDate = (date: string, format: string = FORMAT_DATE) => {
  return moment(date).format(format)
}

export const formatDateNoHours = (date: string, format: string = FORMATE_DATE_NO_HOURS) => {
  return moment(date).format(format)
}
