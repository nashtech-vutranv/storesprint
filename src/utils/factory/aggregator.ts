import faker from '@faker-js/faker'

export const mockAggregator = () => {
  return {
    id: faker.random.word(),
    name: faker.random.word(),
    version: faker.datatype.number(),
    status: faker.datatype.string(),
  }
}

export const mockAggregators = () => {
  return {
    data: {
      content: Array.apply(null, Array(10)).map(() => mockAggregator()),
      totalElements: 10,
    },
  }
}
