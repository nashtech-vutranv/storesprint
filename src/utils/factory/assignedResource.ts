import faker from '@faker-js/faker'

export const mockAssignedResource = () => {
  return {
    data: {
      createdAt: faker.datatype.string(),
      id: faker.datatype.string(),
      modifiedAt: faker.datatype.string(),
      organizationResponse: {
        additionalInformation: faker.datatype.string(),
        addressLine1: faker.datatype.string(),
        addressLine2: null,
        addressLine3: null,
        city: faker.datatype.string(),
        contactEmailAddress: faker.datatype.string(),
        contactName: faker.datatype.string(),
        contactPhoneNumber: faker.datatype.string(),
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        postCode: faker.datatype.string(),
        registrationNumber: faker.datatype.string(),
        status: faker.datatype.string(),
        vatRegistrationNumber: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      siteResponse: {
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        organizationId: faker.datatype.string(),
        status: faker.datatype.string(),
        url: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      userResponse: null,
      version: faker.datatype.number(),
    },
  }
}
