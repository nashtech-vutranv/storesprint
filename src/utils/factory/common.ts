import {IPaginationResponse} from '../../interface'

export const mockOption = () => ({
  value: '',
  label: '',
})

export const mockPaginationResponse = (): IPaginationResponse => {
  return {
    pageable: {
      sort: {
        empty: true,
        sorted: false,
        unsorted: true,
      },
      offset: 0,
      pageSize: 3,
      pageNumber: 0,
      unpaged: false,
      paged: true,
    },
    last: false,
    totalPages: 2,
    totalElements: 5,
    number: 0,
    size: 3,
    first: true,
    numberOfElements: 3,
    sort: {
      empty: true,
      sorted: false,
      unsorted: true,
    },
    empty: false,
  }
}
