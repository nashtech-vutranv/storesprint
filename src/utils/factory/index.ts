export * from './organization'
export * from './warehouse'
export * from './aggregator'
export * from './marketplaces'
export * from './currency'
export * from './relationshipAM'
export * from './marketplaceRelationships'
export * from './product'
export * from './order'
export * from './listingStatus'
export * from './common'
export * from './defaultPropertyValue'
export * from './mappings'
export * from './productProperties'
export * from './assignedResource'
export * from './productCategoryMappings'
export * from './productPropertyMappings'
export * from './courier'
export * from './valueMapping'
export * from './courierMappings'
export * from './returnsrefunds'
export * from './deliveryTypeMappings'
