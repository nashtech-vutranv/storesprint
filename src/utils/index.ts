export * from './array'
export * from './layout'
export * from './date'
export * from './conditionPromise'
