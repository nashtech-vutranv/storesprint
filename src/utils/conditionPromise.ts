import {mockOrganizations, mockSites, mockSiteLocales} from './factory/organization'
import {mockMarketplaces, mockCalculateAssignMarketplaces} from './factory/marketplaces'
import {mockAssignedProducts, mockAssignedProductsResponse} from './factory/product'
import {mockWarehouses} from './factory/warehouse'
import {mockMarketplaceRelationship} from './factory/marketplaceRelationships'

type IMockType = 'organization' | 'site' | 'siteLocale' |
 'marketplace' | 'calculateMarketplace' | 'warehouse' 
 | 'marketplaceRelationship' | 'assignedProduct' | 'assignedProductResponse'

export const conditionPromise = (objectFilter: unknown, typeMock: IMockType, objectPaginationQuery?: any) => {
  if (objectFilter || (objectFilter && objectPaginationQuery)) {
    switch (typeMock) {
      case 'organization': return Promise.resolve(mockOrganizations())
      case 'site': return Promise.resolve(mockSites())
      case 'siteLocale': return Promise.resolve(mockSiteLocales())
      case 'marketplace': return Promise.resolve(mockMarketplaces())
      case 'calculateMarketplace': return Promise.resolve(mockCalculateAssignMarketplaces())
      case 'warehouse': return mockWarehouses()
      case 'marketplaceRelationship': return mockMarketplaceRelationship()
      case 'assignedProduct': return Promise.resolve(mockAssignedProducts())
      case 'assignedProductResponse': return Promise.resolve(mockAssignedProductsResponse())

      default: return {}
    }
  } else return Promise.reject()
}