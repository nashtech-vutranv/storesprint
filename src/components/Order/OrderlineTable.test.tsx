import {render, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import {mockOrderLines} from '../../utils/factory'
import OrderlineTable from './OrderlineTable'

const tree = render(
  <MemoryRouter>
    <OrderlineTable />
  </MemoryRouter>
)

describe('Orderline Table', () => {
  it('should match snapshot with successful data from api', async () => {
    jest.mock('../../services/OrderService', () => {
      const instance = jest.fn().mockReturnValue({
        getOrderlines: jest.fn().mockImplementation(() => {
          return Promise.resolve(mockOrderLines())
        }),
      })

      return instance
    })

    await waitFor(() => {
      expect(tree).toMatchSnapshot()
    })
  })

  it('should match snapshot with error from API', async () => {
    jest.mock('../../services/OrderService', () => {
      const instance = jest.fn().mockReturnValue({
        getOrderlines: jest.fn().mockImplementation(() => {
          return Promise.reject()
        }),
      })

      return instance
    })

    await waitFor(() => {
      expect(tree).toMatchSnapshot()
    })
  })
})
