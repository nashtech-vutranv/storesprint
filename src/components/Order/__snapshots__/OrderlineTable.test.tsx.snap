// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orderline Table should match snapshot with error from API 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body />,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Orderline Table should match snapshot with successful data from api 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-datatable p-component p-datatable-selectable p-datatable-responsive-scroll data-table-mh mt-4"
        data-scrollselectors=".p-datatable-wrapper"
        pr_id_1=""
      >
        <div
          class="p-datatable-wrapper"
        >
          <table
            class="p-datatable-table"
            role="table"
          >
            <thead
              class="p-datatable-thead"
            >
              <tr
                role="row"
              >
                <th
                  class=""
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      order_detail_table_product_erp_id
                    </span>
                  </div>
                </th>
                <th
                  class=""
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      order_detail_table_name
                    </span>
                  </div>
                </th>
                <th
                  class=""
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      order_detail_table_quantity
                    </span>
                  </div>
                </th>
                <th
                  class=""
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      order_detail_table_price
                    </span>
                  </div>
                </th>
                <th
                  class=""
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      order_detail_table_tax
                    </span>
                  </div>
                </th>
                <th
                  class=""
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      order_detail_table_orderline_status
                    </span>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="p-datatable-tbody"
            >
              <tr
                class="p-datatable-emptymessage"
                role="row"
              >
                <td
                  colspan="6"
                  role="cell"
                >
                  order_detail_table_no_data
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="p-paginator p-component p-paginator-bottom table-paginator"
        >
          <div
            class="p-paginator-left-content"
          >
            <button
              aria-label="refresh-button"
              class="p-button p-component p-button-text refresh-button p-button-icon-only"
              type="button"
            >
              <span
                class="p-button-icon p-c pi pi-refresh"
              />
              <span
                class="p-button-label p-c"
              >
                 
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-datatable p-component p-datatable-selectable p-datatable-responsive-scroll data-table-mh mt-4"
      data-scrollselectors=".p-datatable-wrapper"
      pr_id_1=""
    >
      <div
        class="p-datatable-wrapper"
      >
        <table
          class="p-datatable-table"
          role="table"
        >
          <thead
            class="p-datatable-thead"
          >
            <tr
              role="row"
            >
              <th
                class=""
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                >
                  <span
                    class="p-column-title"
                  >
                    order_detail_table_product_erp_id
                  </span>
                </div>
              </th>
              <th
                class=""
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                >
                  <span
                    class="p-column-title"
                  >
                    order_detail_table_name
                  </span>
                </div>
              </th>
              <th
                class=""
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                >
                  <span
                    class="p-column-title"
                  >
                    order_detail_table_quantity
                  </span>
                </div>
              </th>
              <th
                class=""
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                >
                  <span
                    class="p-column-title"
                  >
                    order_detail_table_price
                  </span>
                </div>
              </th>
              <th
                class=""
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                >
                  <span
                    class="p-column-title"
                  >
                    order_detail_table_tax
                  </span>
                </div>
              </th>
              <th
                class=""
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                >
                  <span
                    class="p-column-title"
                  >
                    order_detail_table_orderline_status
                  </span>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="p-datatable-tbody"
          >
            <tr
              class="p-datatable-emptymessage"
              role="row"
            >
              <td
                colspan="6"
                role="cell"
              >
                order_detail_table_no_data
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="p-paginator p-component p-paginator-bottom table-paginator"
      >
        <div
          class="p-paginator-left-content"
        >
          <button
            aria-label="refresh-button"
            class="p-button p-component p-button-text refresh-button p-button-icon-only"
            type="button"
          >
            <span
              class="p-button-icon p-c pi pi-refresh"
            />
            <span
              class="p-button-label p-c"
            >
               
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
